bitnami/spark/templates/deployment-master.yaml

```diff
@@ -0,0 +1,172 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "spark.fullname" . }}-master
+  labels:
+    app.kubernetes.io/name: {{ include "spark.name" . }}
+    helm.sh/chart: {{ include "spark.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/component: master
+spec:
+  replicas: 1
+  strategy:
+    type: {{ .Values.master.updateStrategy }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "spark.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+      app.kubernetes.io/component: master
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "spark.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        helm.sh/chart: {{ include "spark.chart" . }}
+        app.kubernetes.io/component: master
+    {{- if .Values.master.nodeSelector }}
+      nodeSelector:
+      {{ toYaml .Values.master.nodeSelector | nindent 8 }}
```

Comment:
Two options:

Either

```
      nodeSelector: {{ toYaml .Values.master.nodeSelector | nindent 8 }}
```

or

```
      nodeSelector:
      {{- toYaml .Values.master.nodeSelector | nindent 8 }}
```

This way we avoid unnecessary whitespace. Apply it to the rest

Reply:
Done